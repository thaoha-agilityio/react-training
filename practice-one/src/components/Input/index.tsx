import React from 'react';

interface IProps extends React.ComponentPropsWithoutRef<'input'> {
  name?: string;
  value?: string;
  className?: string;
  placeholder?: string;
  type: 'text' | 'checkbox';
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

const Input = ({ name, value, type, onChange, className, placeholder, ...rest }: IProps) => (
  <>
    <input
      name={name}
      value={value}
      onChange={onChange}
      type={type}
      className={className}
      placeholder={placeholder}
      {...rest}
    />
  </>
);

export default Input;
