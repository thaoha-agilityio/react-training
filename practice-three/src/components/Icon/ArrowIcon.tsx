import { SVGProps } from 'react';

export const ArrowIcon = (props: SVGProps<SVGSVGElement>) => (
  <svg width={8} height={11} fill="none" xmlns="http://www.w3.org/2000/svg" {...props}>
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M.147 7.146a.5.5 0 0 1 .708 0L3.5 9.793l2.646-2.647a.5.5 0 0 1 .708.708l-3 3a.5.5 0 0 1-.708 0l-3-3a.5.5 0 0 1 0-.708Z"
      fill="#000000"
    />
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M3.5 0a.5.5 0 0 1 .5.5v9a.5.5 0 1 1-1 0v-9a.5.5 0 0 1 .5-.5Z"
      fill="#000000"
    />
  </svg>
);
