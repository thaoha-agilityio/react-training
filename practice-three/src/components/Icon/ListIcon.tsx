import { SVGProps } from "react";

export const ListIcon = (props: SVGProps<SVGSVGElement>) => (
  <svg
    width={14}
    height={10}
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M4 9a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 9Zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 5Zm0-4a.5.5 0 0 1 .5-.5h9a.5.5 0 0 1 0 1h-9A.5.5 0 0 1 4 1ZM1 2a1 1 0 1 0 0-2 1 1 0 0 0 0 2Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Zm0 4a1 1 0 1 0 0-2 1 1 0 0 0 0 2Z"
      fill="#000"
    />
  </svg>
);
