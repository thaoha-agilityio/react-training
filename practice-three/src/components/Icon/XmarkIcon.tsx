import { SVGProps } from "react";

export const XmarkIcon = (props: SVGProps<SVGSVGElement>) => (
  <svg
    width={14}
    height={14}
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M.22.239a.75.75 0 0 1 1.06 0l5.22 5.22 5.22-5.22a.75.75 0 1 1 1.06 1.06l-5.22 5.22 5.22 5.22a.75.75 0 1 1-1.06 1.06L6.5 7.579l-5.22 5.22a.75.75 0 0 1-1.06-1.06l5.22-5.22-5.22-5.22a.75.75 0 0 1 0-1.06Z"
      fill="#58667E"
    />
  </svg>
);
