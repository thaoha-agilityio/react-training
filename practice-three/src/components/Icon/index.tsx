export * from "./ArrowIcon";

export * from "./FilterIcon";

export * from "./GridIcon";

export * from "./LightIcon";

export * from "./ListIcon";

export * from "./DarkIcon";

export * from "./SearchIcon";

export * from "./XmarkIcon";

export * from "./ArrowRightIcon";
