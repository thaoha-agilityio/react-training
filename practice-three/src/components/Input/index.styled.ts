import styled from "styled-components";

import { metrics, fonts, colors } from "../../themes";

interface StyledInputProps {
  borderRadius: number;
  pLeft: number;
  width?: number;
  height?: number;
}

const InputGroupStyled = styled.div`
  display: flex;
  position: relative;
`;

const InputStartAdornmentStyled = styled.div`
  position: absolute;
  z-index: 1;
  padding-top: ${metrics.paddings.md}px;
  padding-left: ${metrics.paddings.lg}px;
`;

const InputStyled = styled.input<StyledInputProps>`
  ${(props) => `width:${props.width}px`};
  ${(props) => `height:${props.height}px`};
  ${(props) => `border-radius:${props.borderRadius}px`};
  ${(props) => `padding-left:${props.pLeft}px`};
  font-family: ${fonts.fontFamilies.primary};
  box-shadow: ${metrics.shadows.sm};
  font-size: ${fonts.fontSizes.sm}px;
  line-height: ${fonts.lineHeights.tiny};
  border: none;

  &::placeholder {
    color: ${colors.ash};
  }
`;

export { InputGroupStyled, InputStartAdornmentStyled, InputStyled };
