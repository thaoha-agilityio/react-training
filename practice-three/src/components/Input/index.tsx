import { ComponentPropsWithRef, ReactNode } from "react";

import {
  InputGroupStyled,
  InputStartAdornmentStyled,
  InputStyled,
} from "./index.styled";

interface IProps extends Omit<ComponentPropsWithRef<"input">, "size"> {
  width?: number;
  height?: number;
  borderRadius: number;
  pLeft: number;
  children?: ReactNode;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

const Input = ({
  width = 670,
  height = 53,
  borderRadius,
  pLeft,
  children,
  onChange,
  ...rest
}: IProps) => {
  return (
    <InputGroupStyled>
      <InputStartAdornmentStyled children={children} />
      <InputStyled
        width={width}
        height={height}
        onChange={onChange}
        borderRadius={borderRadius}
        pLeft={pLeft}
        {...rest}
      />
    </InputGroupStyled>
  );
};

export default Input;
