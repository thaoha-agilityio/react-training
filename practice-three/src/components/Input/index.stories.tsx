import { ComponentStory, ComponentMeta } from "@storybook/react";

import { SearchIcon } from "@/components/Icon";

import Input from ".";

export default {
  title: "Input",
  component: Input,
  argTypes: {
    isCircle: {
      defaultValue: false,
      control: { type: "boolean" },
    },
    children: {
      defaultValue: <SearchIcon />,
    },
    width: {
      defaultValue: 670,
    },
    height: {
      defaultValue: 53,
    },
    pLeft: {
      defaultValue: 48,
    },
    borderRadius: {
      defaultValue: 30,
    },
    placeholder: {
      defaultValue: "Search books",
    },
  },
  parameters: {},
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />;

export const Basic = Template.bind({});
