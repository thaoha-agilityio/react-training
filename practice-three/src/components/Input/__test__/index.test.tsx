import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

import { SearchIcon } from "../../Icon";

import Input from "..";

const mockProps = {
  type: "text",
  width: 670,
  height: 53,
  borderRadius: 30,
  pLeft: 48,
  children: <SearchIcon />,
  placeholder: "Search book",
};

const setupGetValue = () => {
  const component = render(<Input {...mockProps} />);
  const input = component.getByPlaceholderText(
    "Search book"
  ) as HTMLInputElement;
  return {
    input,
    component,
  };
};

describe("Input render", () => {
  it("Should match Input DOM Snapshot", () => {
    const { input, component } = setupGetValue();

    expect(input).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it("Should be able type in input", () => {
    const { input } = setupGetValue();
    fireEvent.change(input, { target: { value: "Angels and demons" } });
    expect(input.value).toBe("Angels and demons");
  });
});
