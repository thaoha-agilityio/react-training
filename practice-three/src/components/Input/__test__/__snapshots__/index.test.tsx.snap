// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render Should match Input DOM Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe eDetyh"
      >
        <div
          class="sc-gswNZR keBqvU"
        >
          <svg
            fill="none"
            height="22"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.38 18.143c4.629 0 8.38-3.838 8.38-8.572C17.76 4.838 14.01 1 9.38 1 4.752 1 1 4.838 1 9.571c0 4.734 3.752 8.572 8.38 8.572ZM15.666 16l4.888 5"
              stroke="#B8B9BE"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <input
          class="sc-dkrFOg qLnSR"
          height="53"
          placeholder="Search book"
          type="text"
          width="670"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe eDetyh"
    >
      <div
        class="sc-gswNZR keBqvU"
      >
        <svg
          fill="none"
          height="22"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.38 18.143c4.629 0 8.38-3.838 8.38-8.572C17.76 4.838 14.01 1 9.38 1 4.752 1 1 4.838 1 9.571c0 4.734 3.752 8.572 8.38 8.572ZM15.666 16l4.888 5"
            stroke="#B8B9BE"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <input
        class="sc-dkrFOg qLnSR"
        height="53"
        placeholder="Search book"
        type="text"
        width="670"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`InputShop render Should match Input DOM Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe eDetyh"
      >
        <div
          class="sc-gswNZR keBqvU"
        >
          <svg
            fill="none"
            height="22"
            width="22"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.38 18.143c4.629 0 8.38-3.838 8.38-8.572C17.76 4.838 14.01 1 9.38 1 4.752 1 1 4.838 1 9.571c0 4.734 3.752 8.572 8.38 8.572ZM15.666 16l4.888 5"
              stroke="#B8B9BE"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <input
          class="sc-dkrFOg qLnSR"
          height="53"
          placeholder="Search book"
          type="text"
          width="670"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe eDetyh"
    >
      <div
        class="sc-gswNZR keBqvU"
      >
        <svg
          fill="none"
          height="22"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.38 18.143c4.629 0 8.38-3.838 8.38-8.572C17.76 4.838 14.01 1 9.38 1 4.752 1 1 4.838 1 9.571c0 4.734 3.752 8.572 8.38 8.572ZM15.666 16l4.888 5"
            stroke="#B8B9BE"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <input
        class="sc-dkrFOg qLnSR"
        height="53"
        placeholder="Search book"
        type="text"
        width="670"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
